src/main/java/de/bonndan/nivio/security/CustomOAuth2UserService.java

```diff
@@ -46,30 +48,30 @@ public static CustomOAuth2User fromGitHubUser(@NonNull final OAuth2User user,
                                                   @Nullable final String aliasAttribute,
                                                   @Nullable final String nameAttribute
     ) {
-        var id = "";
+        var externalId = "";
         if (StringUtils.hasLength(nameAttribute)) {
-            id = String.valueOf(user.getAttribute("id") == null ? "" : user.getAttribute("id"));
+            externalId = String.valueOf(user.getAttribute("id") == null ? "" : user.getAttribute("id"));
         }
 
         var name = "";
         if (StringUtils.hasLength(nameAttribute)) {
             Object val = user.getAttribute(nameAttribute);
             if (val == null) {
-                Object login = Objects.requireNonNull(user.getAttribute("login"));
+                Object login = Objects.requireNonNull(user.getAttribute(aliasAttribute));
                 name = String.valueOf(login);
             } else {
                 name = String.valueOf(val);
             }
         }
 
         return new CustomOAuth2User(
-                id,
+                externalId,
                 StringUtils.hasLength(aliasAttribute) ? Optional.ofNullable((String) user.getAttribute(aliasAttribute)).orElse("") : "",
                 name,
                 user.getAttributes(),
                 user.getAuthorities(),
-                user.getAttribute("avatar_url")
-        );
+                user.getAttribute("avatar_url"),
+                "github");
```

Comment:
Avoid hard coded values.

Reply:
Good point: I think we leave it as it is and keep that in mind when adding another oauth2 implementation.